test/unittest/transport/UDPv6Tests.cpp

```diff
@@ -89,15 +103,129 @@ TEST_F(UDPv6Tests, conversion_to_ip6_string)
 
     locator.address[5] = 0x0c;
     ASSERT_EQ("ffaa:a00:c::", IPLocator::toIPv6string(locator));
+
+    //Zero Compression of 2001:DB8:0:0:8:800:200C:417A
+    reset_locator_address(locator);
+    locator.address[0] = 0x20;
+    locator.address[1] = 0x01;
+    locator.address[2] = 0x0d;
+    locator.address[3] = 0xb8;
+    locator.address[9] = 0x08;
+    locator.address[10] = 0x08;
+    locator.address[12] = 0x20;
+    locator.address[13] = 0x0c;
+    locator.address[14] = 0x41;
+    locator.address[15] = 0x7a;
+    ASSERT_EQ("2001:db8::8:800:200c:417a", IPLocator::toIPv6string(locator));
+
+    //Zero Compression of FF01:0:0:0:0:0:0:101
+    reset_locator_address(locator);
+    locator.address[0] = 0xff;
+    locator.address[1] = 0x01;
+    locator.address[14] = 0x01;
+    locator.address[15] = 0x01;
+    ASSERT_EQ("ff01::101", IPLocator::toIPv6string(locator));
+
+    //Zero Compression of 0:0:0:0:0:0:0:1
+    reset_locator_address(locator);
+    locator.address[15] = 0x01;
+    ASSERT_EQ("::1", IPLocator::toIPv6string(locator));
+
+    //Zero Compression of 0:0:0:0:0:0:0:0
+    reset_locator_address(locator);
+    locator.address[15] = 0x00;
+    ASSERT_EQ("::", IPLocator::toIPv6string(locator));
+
+    //Trailing Zeros
+    reset_locator_address(locator);
+    locator.address[14] = 0x10;
+    ASSERT_EQ("::1000", IPLocator::toIPv6string(locator));
+    locator.address[15] = 0x20;
+    ASSERT_EQ("::1020", IPLocator::toIPv6string(locator));
+
+    //Embedded Zeros in 2001:DB8:a::/48
+    reset_locator_address(locator);
```

Comment:
Partial review.
I have reviewed the tests, they are very thorough and cover the all the cases of address compression. A nitpick/question, is this `/48` a typo?

Feel free to commit the fix when you want, and I will continue the review.

Reply:
Yes copy and paste from the RFC that is not relevant. Will fix